ui/src/components/editor/shapeComponents/setDefinition/QuickQueryDialog.vue

```diff
@@ -0,0 +1,247 @@
+import { EntityService, QueryService } from "@/services";
+import { isArrayHasLength, isObject, isObjectHasKeys } from "@im-library/helpers/DataTypeCheckers";
+import { ConceptSummary, FilterOptions } from "@im-library/interfaces";
+import { useFilterStore } from "@/stores/filterStore";
+import setupDownloadFile from "@/composables/downloadFile";
+import { ToastOptions } from "@im-library/models";
+import { ToastSeverity } from "@im-library/enums";
+import { useToast } from "primevue/usetoast";
+import { IM, RDFS, SHACL, XSD } from "@im-library/vocabulary";
+import { cloneDeep } from "lodash";
+
+interface Props {
+  queryIri?: string;
+  query?: string;
+  showDialog: boolean;
+}
+
+const props = defineProps<Props>();
+
+const filterStore = useFilterStore();
+const controller: Ref<AbortController> = ref({} as AbortController);
+const queryLoading: Ref<boolean> = ref(false);
+const debounce = ref(0);
+const filterDefaults: ComputedRef<FilterOptions> = computed(() => filterStore.filterDefaults);
+const suggestions: Ref<ConceptSummary[]> = ref([]);
+const toast = useToast();
+const visible = ref(false);
+const { downloadFile } = setupDownloadFile(window, document);
+const params: Ref<{ name: string; desc: string; type: string; minCount: number; maxCount: number; value: any }[]> = ref([]);
+const queryResults: Ref<{ entities: any[]; "@context": any }> = ref({} as { entities: any[]; "@context": any });
+const cols: Ref<{ field: string }[]> = ref([]);
+
+const emit = defineEmits({
+  "update:showDialog": payload => typeof payload === "boolean"
+});
+
+watch(
+  () => props.showDialog,
+  newValue => {
+    visible.value = newValue;
+  }
+);
+
+watch(visible, newValue => {
+  if (!newValue) {
+    emit("update:showDialog", newValue);
+  }
+});
+
+watch(
+  () => props.queryIri,
+  async () => await init()
+);
+
+watch(
+  () => cloneDeep(props.query),
+  async () => await init()
+);
+
+onMounted(async () => await init());
+
+async function init() {
+  params.value = await getParams();
+}
+
+function clearResults() {
+  queryResults.value = { entities: [], "@context": {} };
+}
+
+async function getQueryRequest() {
+  const request = {} as QueryRequest;
+
+  if (props.queryIri) {
+    const entity = await EntityService.getPartialEntity(props.queryIri, [IM.DEFINITION]);
+    if (isObjectHasKeys(entity, [IM.DEFINITION])) request.query = JSON.parse(entity[IM.DEFINITION]);
+  } else if (props.query) {
+    request.query = JSON.parse(props.query);
+  }
+
+  return request;
+}
+
+function addArguments(queryRequest: QueryRequest) {
+  queryRequest.argument = [];
+  for (const param of params.value) {
+    if ("Text search" === param.name) {
+      if (param.value) queryRequest.textSearch = param.value;
+    } else {
+      const argument = {
+        parameter: param.name
+      } as Argument;
+
+      if (isArrayHasLength(param.value)) {
+        argument.valueIriList = (param.value as []).map((summary: ConceptSummary) => {
```

Comment:
if param.value is always ConceptSummary update ref typing at declaration. The no need to cast types.

Otherwise `param.value as ConceptSummary[]` single casting.